### PR TITLE
Stable diffusion webui support

### LINKED ISSUE #3: Support for stable-diffusion-webui
Add support for [stable-diffusion-webui](https://github.com/AUTOMATIC1111/stable-diffusion-webui).
stable-diffusion-webui has far more features and community support than InvokeAI does.

### PR DESCRIPTION
**Pull request by RolfNic**
Adds support for the [Stable Diffusion Webui](https://github.com/AUTOMATIC1111/stable-diffusion-webui), which has more features and community support than InvokeAI. resolves #3.

### CODE REVIEW

File: packages/lpips/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+{ buildPythonPackage, fetchPypi, lib }:
+
+buildPythonPackage rec {
+  pname = "lpips";
+  version = "0.1.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0afxbg5scrlydqkm1cwlnlb0mffrzl785gn4wxrw9xnpqmn130fl";
```
Comment (RolfNic) [inline packages/lpips/default.nix]: this hash is incorrect and needs to be updated

**Comment by RolfNic** [comment]
Text2Img works on my machine, img2img is still broken

File: packages/markdown-it-py/default.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    rev = "refs/tags/v${version}";
+    sha256 = "sha256-6UATJho3SuIbLktZtFcDrCTWIAh52E+n5adcgl49un0=";
+  };
+
+  postPatch = ''
+    # Allow linkify-it-py v2.
+    # This only affects projects that depend on 'markdown-it-py[linkify]'
+    # https://github.com/executablebooks/markdown-it-py/pull/218
+    substituteInPlace pyproject.toml \
+      --replace "linkify-it-py~=1.0" "linkify-it-py>=1,<3"
```
Comment (gbtb) [inline packages/markdown-it-py/default.nix:35]: You can use pythonRelaxDepsHook Instead of manual version replace

File: .gitmodules [hunk 3]
```diff
@@ -2,3 +2,6 @@
 	path = InvokeAI
 	url = https://github.com/invoke-ai/InvokeAI
 	branch = main
+[submodule "stable-diffusion-webui"]
```
Comment (gbtb) [inline .gitmodules]: There is an incompatibility between https clone urls and ssh ones - I pulled your branch and was not able to update new submodule. Relative urls make it compatible with both - https://www.damirscorner.com/blog/posts/20210423-ChangingUrlsOfGitSubmodules.html

```suggestion
[submodule "InvokeAI"]
     path = InvokeAI
     url = ../../invoke-ai/InvokeAI
     branch = main
 [submodule "stable-diffusion-webui"]
     path = stable-diffusion-webui
     url = ../../AUTOMATIC1111/stable-diffusion-webui
```

**Comment by gbtb** [comment]
> Text2Img works on my machine, img2img is still broken

Looking into https://github.com/AUTOMATIC1111/stable-diffusion-webui/blob/master/launch.py, I'm not surprised that something is not yet working :wink: . I'm just glad I started with InvokeAI, because it's been a lot simpler to package - all dependencies were declared in appropriate files, and I've wasted more time fighting with ROCM for torch and that opencv-blas segfault than actually packaging missing deps.

**Comment by gbtb** [comment]
I've tested your branch and also experienced the same problem with img2img. Then I checked browser console for errors and it led me to this [issue](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/861). Looks like we need to upgrade gradio to at least version 3.3.1, though in requirements.txt it explicitly set to 3.5

**Comment by RolfNic** [comment]
> I've tested your branch and also experienced the same problem with img2img. Then I checked browser console for errors and it led me to this [issue](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/861). Looks like we need to upgrade gradio to at least version 3.3.1, though in requirements.txt it explicitly set to 3.5

Yea I should've probably mentioned this, but while webui uses gradio 3.6 I couldn't package anything over 3.3 so I just decreased the version number till it worked. The error I ran into:

```
Executing setuptoolsBuildPhase
Traceback (most recent call last):
  File "/build/gradio-3.6/nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "/nix/store/wyhbl43ycqn43d08v5fqj1j6ynf7nz73-python3-3.10.7/lib/python3.10/tokenize.py", line 394, in open
    buffer = _builtin_open(filename, 'rb')
FileNotFoundError: [Errno 2] No such file or directory: 'setup.py'
```

**Comment by gbtb** [comment]
> > I've tested your branch and also experienced the same problem with img2img. Then I checked browser console for errors and it led me to this [issue](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/861). Looks like we need to upgrade gradio to at least version 3.3.1, though in requirements.txt it explicitly set to 3.5
> 
> Yea I should've probably mentioned this, but while webui uses gradio 3.6 I couldn't package anything over 3.3 so I just decreased the version number till it worked. The error I ran into:
> 
> ```
> Executing setuptoolsBuildPhase
> Traceback (most recent call last):
>   File "/build/gradio-3.6/nix_run_setup", line 8, in <module>
>     exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
>   File "/nix/store/wyhbl43ycqn43d08v5fqj1j6ynf7nz73-python3-3.10.7/lib/python3.10/tokenize.py", line 394, in open
>     buffer = _builtin_open(filename, 'rb')
> FileNotFoundError: [Errno 2] No such file or directory: 'setup.py'
> ```

Yes, they've removed setup.py exactly between version 3.3 and 3.3.1 :) Btw, I've continued working on this branch on a weekend, I think all that's left is to package BLIP model and sync branch with master. You can test my latest changes from the [PR](https://github.com/RolfNic/nix-stable-diffusion/pull/1) I've made to your repo.

**Comment by gbtb** [comment]
Hey, I've finished this branch and stable-diffusion-webui support is now merged into master. Thus, I'm closing this PR. Thanks again for your help :)

**Pull request closed without merging**